src/explore-education-statistics-common/src/components/__tests__/Tooltip.test.tsx

```diff
@@ -0,0 +1,251 @@
+    render(
+      <Tooltip text="Test tooltip" id="test-tooltip" enabled={false}>
+        {({ ref }) => (
+          <button ref={ref} type="button">
+            Test button
+          </button>
+        )}
+      </Tooltip>,
+    );
+
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+  });
+
+  test('links reference element to tooltip via `aria-describedby`', () => {
+    render(
+      <Tooltip text="Test tooltip" id="test-tooltip">
+        {({ ref }) => (
+          <button ref={ref} type="button">
+            Test button
+          </button>
+        )}
+      </Tooltip>,
+    );
+
+    expect(screen.getByRole('tooltip', { hidden: true })).toHaveAttribute(
+      'id',
+      'test-tooltip',
+    );
+    expect(screen.getByRole('button', { name: 'Test button' })).toHaveAttribute(
+      'aria-describedby',
+      'test-tooltip',
+    );
+  });
+
+  test('links reference element with existing `aria-describedby` to tooltip', () => {
+    render(
+      <Tooltip text="Test tooltip" id="test-tooltip">
+        {({ ref }) => (
+          <button aria-describedby="something-else" ref={ref} type="button">
+            Test button
+          </button>
+        )}
+      </Tooltip>,
+    );
+
+    expect(screen.getByRole('tooltip', { hidden: true })).toHaveAttribute(
+      'id',
+      'test-tooltip',
+    );
+    expect(screen.getByRole('button', { name: 'Test button' })).toHaveAttribute(
+      'aria-describedby',
+      'test-tooltip something-else',
+    );
+  });
+
+  test('clicking reference element makes tooltip visible', () => {
+    render(
+      <Tooltip text="Test tooltip">
+        {({ ref }) => (
+          <button ref={ref} type="button">
+            Test button
+          </button>
+        )}
+      </Tooltip>,
+    );
+
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+    userEvent.click(screen.getByRole('button', { name: 'Test button' }));
+
+    expect(screen.getByRole('tooltip')).toBeInTheDocument();
+  });
+
+  test('hovering reference element makes tooltip visible', () => {
+    render(
+      <Tooltip text="Test tooltip">
+        {({ ref }) => (
+          <button ref={ref} type="button">
+            Test button
+          </button>
+        )}
+      </Tooltip>,
+    );
+
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+    userEvent.hover(screen.getByRole('button', { name: 'Test button' }));
+
+    expect(screen.getByRole('tooltip')).toBeInTheDocument();
+  });
+
+  test('hovering reference element makes tooltip invisible', () => {
+    render(
+      <Tooltip text="Test tooltip">
+        {({ ref }) => (
+          <button ref={ref} type="button">
+            Test button
+          </button>
+        )}
+      </Tooltip>,
+    );
+
+    expect(screen.queryByRole('tooltip')).not.toBeInTheDocument();
+
+    userEvent.hover(screen.getByRole('button', { name: 'Test button' }));
+
+    expect(screen.getByRole('tooltip')).toBeInTheDocument();
+  });
```

Comment:
This seems to be the same test as the one before apart from the description - visible / invisible.